scene/main/viewport.cpp

```diff
@@ -1935,19 +1935,18 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 
 				// If the tooltip timer isn't running, start it.
 				// Otherwise, only reset the timer if the mouse has moved more than 5 pixels.
-				if (!is_tooltip_shown && over->can_process() &&
-						(gui.tooltip_timer.is_null() ||
-								Math::is_zero_approx(gui.tooltip_timer->get_time_left()) ||
```

Comment:
I don't think we want to remove the check for a stopped timer. The goal of the minimum distance test is to prevent the timer from from being reset by small mouse movements, _not_ to prevent the timer from being started in the first place. With the proposed change, a user may be confused by a tooltip not displaying just because they haven't moved their mouse enough. This could happen if their mouse had been sitting just a few pixels away from the control of interest before they moved to it.

Also, the proposed change no longer matches the comments describing the desired behavior above.

Reply:
I see. I think checking for the stopped timer also doesn't resolve that problem. Eg. mouse is moved to near the control, then it's moved inside the control before the timer stops => no triggering.

I think the solution is to check if there's a new control under the mouse and restart the timer in that case.

Reply:
Yes, we could reset the timer if `over` doesn't match `gui.tooltip_control`.